envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/SnapshotProperties.kt

```diff
@@ -62,6 +62,13 @@ class IncomingPermissionsProperties {
     var enabled = false
     var clientIdentityHeader = "x-service-name"
     var sourceIpAuthentication = SourceIpAuthenticationProperties()
+    var tlsAuthentication = TlsAuthenticationProperties()
+}
+
+class TlsAuthenticationProperties {
+    val requireClientCertificate: Boolean = true // if false, will not validate cert
```

Comment:
Why do we want to enable mtls by default ? It is safer if deploy of newer version won't change behaviour without our knowledge.

Reply:
An envoy has to have metadata info to have the listener enabled so our migration process is gradual. I treat this more as a "kill switch" in case we need it. WDYT @MarcinFalkowski ?

Reply:
I think there is no point to NOT require client certificate. We launch mTLS as a second filter chain on the listener. Plain HTTP requests still work, so we don't loose backward compatibility. The only clients which will make TLS requests will be compatible Envoys. Compatible Envoys will always present a client certificate.